### DISCUSSION TITLE
defaultCPUTime

### DISCUSSION THREAD

**Discussion by ernstpijper** [DIRACGrid/DIRAC#7074] [Q&A] [answered]
Hi,

DIRAC py3 server 7.3.17, py3 pilot jobs and py3 diracos2 client

I have set my defaultCPUTime to 345600 seconds (96 hours) but was wondering if it was possible to set the defaultCPUTime based on VO.

Thanks, 
Ernst

**Comment by fstagni** [comment]
Hi Ernst,
I guess you have set it in the `/Operations` section of the Configuration, and in this case it is possible to set it per-VO. Where did you set it?

Cheers,
Federico

**Reply by ernstpijper** [reply]
Hi Federico,

I set it in /Operations/Defaults/JobDescription/DefaultCPUTime = 345600

**Reply by fstagni** [reply, marked as answer]
The you can just set it (additionally) for the VOs for which you want a different `DefaultCPUTime` in `/Operations/vo_name/JobDescription/DefaultCPUTime`

**Comment by ernstpijper** [comment]
Hi Federico,

I guess this is what I tried before and found out it didn't work.  Tried it again and I can see that de defaultcpdutime is still set to the wrong value. My config looks like this:

![Screenshot 2023-06-27 at 10 46 18](https://github.com/DIRACGrid/DIRAC/assets/4125387/d837bb24-1e77-496f-a393-010faac02c50)

![image](https://github.com/DIRACGrid/DIRAC/assets/4125387/f271e74a-a85b-43da-88c6-288ea3c21da0)

**Reply by fstagni** [reply]
Few questions:
- was the job above submitted before or after the change in CS?
- what is the OwnerGroup of the job you show in the screenshot?
- are you sure you are not forcing a CPU time by other means?

**Comment by ernstpijper** [comment]
Hi Federico,

Yes it was submitted after the change. I also did another reload (using runsvctrl) and submitted another job. Same result.
The OwnerGroup is pvier_user. I used the dirac egg naming convention.
I'm not enforcing the cputime anywhere else. I checked and cannot find any another place where this value is set.

**Comment by fstagni** [comment]
Can you try the following from a client?

```python
from DIRAC.Core.Base.Script import parseCommandLine
parseCommandLine()

from DIRAC import gLogger
gLogger.setLevel("DEBUG")

from DIRAC.ConfigurationSystem.Client.Helpers.Operations import Operations

opsHelper = Operations(group="pvier_user")
print(opsHelper.getValue("JobDescription/DefaultCPUTime"))
```
? 

What is the result?

**Reply by ernstpijper** [reply]
Got this, am i doing something wrong:

ernstp$ cat cputime.py 
from DIRAC import initialize
initialize()

opsHelper = Operations(group="pvier_user")
opsHelper.getValue("JobDescription/DefaultCPUTime")
(base) 

Tue Jun 27 14:20 ui.grid.surfsara.nl:/home/ernstp/Grid 
ernstp$ python cputime.py 
Traceback (most recent call last):
  File "/home/ernstp/Grid/cputime.py", line 1, in <module>
    from DIRAC import initialize
ImportError: cannot import name 'initialize' from 'DIRAC' (/usr/local/diracos/lib/python3.9/site-packages/DIRAC/__init__.py)
(base)

**Reply by fstagni** [reply]
ah, sorry, version "issue". Do the slightly modified version above (I edited my previous comment).

**Comment by ernstpijper** [comment]
Still not:

ernstp$ python cputime.py 
Traceback (most recent call last):
  File "/home/ernstp/Grid/cputime.py", line 1, in <module>
    from DIRAC.Core.Base.Script import Script
ImportError: cannot import name 'Script' from 'DIRAC.Core.Base.Script' (/usr/local/diracos/lib/python3.9/site-packages/DIRAC/Core/Base/Script.py)
(base)

**Reply by fstagni** [reply]
Try again (re-edited).

**Comment by ernstpijper** [comment]
ernstp$ python cputime.py 
345600
(base)

**Reply by fstagni** [reply]
I think I see it now: add `/Operations/vo_name/DIRAC-Production/JobDescription/DefaultCPUTime`

**Reply by andresailer** [reply]
Maybe `Defaults` also works: 
```
/Operations/vo_name/Defaults/JobDescription/DefaultCPUTime
                    ^^^^^^^^
```

**Comment by ernstpijper** [comment]
Hi,

This one

/Operations/vo_name/DIRAC-Production/JobDescription/DefaultCPUTime

did not work but this one:

/Operations/vo_name/Defaults/JobDescription/DefaultCPUTime

did work.

Thank for your help.
